packages/mds-web-sockets/clients.ts

```diff
@@ -37,16 +57,41 @@ export class Clients {
     )
   }
 
-  public saveAuth(token: string, client: WebSocket) {
+  public async saveAuth(authorizer: string, client: WebSocket) {
     try {
-      const auth = WebSocketAuthorizer(token)
+      const [, token] = authorizer.split(' ')
+
+      const auth = WebSocketAuthorizer(authorizer)
+
+      const validateAuth = await Clients.checkAuth(token)
+      if (!validateAuth) {
+        client.send(JSON.stringify({ err: new AuthorizationError() }))
+        return
+      }
+
       const scopes = auth?.scope.split(' ') ?? []
+
       if (scopes.includes('admin:all')) {
```

Comment:
Would the scope to check for here be an argument to this function in the future?

Reply:
Absolutely. Depending on how long we keep this package (there's a chance we might dump it in the future when NATS JetStream comes around), I'll need to spec out scopes for the websocket interface.